tests/src/JIT/HardwareIntrinsics/X86/Sse2/AddSaturate.cs

```diff
@@ -0,0 +1,119 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+using System;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+namespace IntelHardwareIntrinsicTest
+{
+    internal static partial class Program
+    {
+        const int Pass = 100;
+        const int Fail = 0;
+
+        static unsafe int Main(string[] args)
+        {
+            int testResult = Pass;
+            int testsCount = 21;
+            string methodUnderTestName = nameof(Sse2.AddSaturate);
+
+            if (Sse2.IsSupported)
+            {
+                using (var shortTable = TestTableSse2<short>.Create(testsCount))
+                using (var ushortTable = TestTableSse2<ushort>.Create(testsCount))
+                using (var sbyteTable = TestTableSse2<sbyte>.Create(testsCount))
+                using (var byteTable = TestTableSse2<byte>.Create(testsCount))
+                {
+                    for (int i = 0; i < testsCount; i++)
```

Comment:
Why the loop for each of these?

Reply:
First of all this is my old test code which was written before we have discussion on how to test HW intrinsics in #15771 about testing in general. In my updated remarks in the OP https://github.com/dotnet/coreclr/pull/15585#issue-283537124 I have indicated that due to decisions made on how we test I have to change this code entirely. Since we plan single project/assembly per ISA with support of T4 code generation my plan is to do it in last commit and generate all tests with common template.

Right now the code runs tests on random data generated in `TestTableSse2` and this requires looping over 21 pairs of vectors of data. Number 21 is chosen based on other arithmetic tests I have seen in coreclr and corefx. Anyway it will not be used in final version of tests.